src/java/org/apache/cassandra/config/Converters.java

```diff
@@ -0,0 +1,122 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.config;
+
+import java.util.function.Function;
+
+/**
+ * Converters for backward compatibility with the old cassandra.yaml where duration, data rate and
+ * data storage configuration parameters were provided only by value and the expected unit was part of the configuration
+ * parameter name(suffix). (CASSANDRA-15234)
+ * It is important to be noted that this converter is not intended to be used when we don't change name of a configuration
+ * parameter but we want to add unit. This would always default to the old value provided without a unit at the moment.
+ * In case this functionality is needed at some point, please, raise a Jira ticket.
+ */
+public enum Converters
+{
+    /**
+     * This converter is used when we change the name of a cassandra.yaml configuration parameter but we want to be
+     * able to still use the old name too. No units involved.
+     */
+    IDENTITY(null, o -> o, o-> o),
+    MILLIS_DURATION(Long.class,
+                    o -> DurationSpec.inMilliseconds((Long) o),
+                    o -> ((DurationSpec)o).toMilliseconds()),
+    MILLIS_DOUBLE_DURATION(Double.class,
+                           o ->  DurationSpec.inDoubleMilliseconds((Double) o),
+                           o -> ((DurationSpec)o).toMilliseconds()),
+    /**
+     * This converter is used to support backward compatibility for parameters where in the past -1 was used as a value
+     * Example: credentials_update_interval_in_ms = -1 and credentials_update_interval = null (quantity of 0ms) are equal.
+     */
+    MILLIS_CUSTOM_DURATION(Long.class,
+                           o -> (Long)o == -1 ? new DurationSpec("0ms") : DurationSpec.inMilliseconds((Long) o),
+                           o -> ((DurationSpec)o).toMilliseconds() == 0 ? -1 : ((DurationSpec)o).toMilliseconds()),
+    SECONDS_DURATION(Long.class,
+                     o -> DurationSpec.inSeconds((Long) o),
+                     o -> ((DurationSpec)o).toSeconds()),
+    MINUTES_DURATION(Long.class,
+                     o -> DurationSpec.inMinutes((Long) o),
+                     o -> ((DurationSpec)o).toMinutes()),
+    MEBIBYTES_DATA_STORAGE(Long.class,
+                          o -> DataStorageSpec.inMebibytes((Long) o),
+                          o -> ((DataStorageSpec)o).toMebibytes()),
+    KIBIBYTES_DATASTORAGE(Long.class,
+                          o -> DataStorageSpec.inKibibytes((Long) o),
+                          o -> ((DataStorageSpec)o).toKibibytes()),
+    BYTES_DATASTORAGE(Long.class,
+                      o -> DataStorageSpec.inBytes((Long) o),
+                      o -> ((DataStorageSpec)o).toBytes()),
+    MEBIBYTES_PER_SECOND_DATA_RATE(Long.class,
+                                   o -> DataRateSpec.inMebibytesPerSecond((Long) o),
```

Comment:
nit: It should matter here, but if we used `Long.valueOf()`, the converters could also handle incoming `String` values.

Reply:
I don't think we need `String` here?

Reply:
Yeah, I wouldn't worry too much about it.